### PR TITLE
Frontend Misc

### PR DESCRIPTION
**Pull request by mf59816**
- user registering with service
- snap error handling
- use uri-bytestring package
- various misc refactorings and bug fixes

### CODE REVIEW

**Comment by mf59816** [comment]
(Travis is expected to fail.  See comment in import section in `Thentos.Frontend.Handlers`.  I'm a bit confused about the error travis actually throws, but I'll look into this in a different PR once uri-bytestring-0.1 is out.)

**Comment by fhartwig** [comment]
@mf59816 The travis error is because we haven't added `uri-bytestring` to our dependency cache, right?

File: src/Thentos/Frontend/Handlers.hs [hunk 1]
```diff
@@ -1,47 +1,56 @@
...
+import Snap.Snaplet.Session (commitSession, setInSession, getFromSession, resetSession)
 import Snap.Snaplet (with)
+import System.Log.Missing (logger)
 import System.Log (Priority(DEBUG, INFO, WARNING, CRITICAL))
 import Text.Digestive.Snap (runForm)
+import URI.ByteString (URI(..), RelativeRef(..), URIParserOptions)
+import URI.ByteString (parseURI, parseRelativeRef, laxURIParserOptions, serializeURI, serializeRelativeRef)
+import URI.ByteString (rrPathL, uriQueryL, queryPairsL)
+
+-- We are using an experimental version of URI.ByteString:
```
Comment (fhartwig) [inline src/Thentos/Frontend/Handlers.hs]: Could we not add this version to our travis dependency cache, since we just copy the entire sandbox anyway?

Comment (mf59816) [inline src/Thentos/Frontend/Handlers.hs]: yes, travis doesn't find this in the cache, but the script was supposed to fall back to installing missings.

I would prefer not to fix this in the travis cache until we have a working release on hackage.  Shouldn't be long: https://github.com/Soostone/uri-bytestring/pull/4

Comment (fhartwig) [inline src/Thentos/Frontend/Handlers.hs]: Ok, we'll leave it for now.

**Pull request merged**